### PR TITLE
Live Activities UI

### PR DESCRIPTION
**Pull request by guytepper**
IMG_7B31D12C94F8-1

### CODE REVIEW

File: app/utils/ios-helpers.tsx [hunk 1]
```diff
@@ -9,3 +10,75 @@ export function donateRouteIntent(originId: string, destinationId: string) {
...
+    modifiedTrain.orignStation = train.originStationId
+
+    // @ts-expect-error
+    modifiedTrain.destinationStation = train.destinationStationId
+
+    modifiedTrain.stopStations = train.stopStations.map((stopStation) => {
+      const modifiedStopStation = { ...stopStation }
+      // @ts-expect-error
+      modifiedStopStation.arrivalTime = stopStation.arrivalTimeString
+      // @ts-expect-error
```
Comment (planecore) [inline app/utils/ios-helpers.tsx:47]: Why all the `// @ts-expect-error`?

File: ios/BetterRailWidget/Live Activity/Views/DynamicIsland/BottomView.swift [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+//  BetterRailWidgetExtension
+//
+//  Created by Guy Tepper on 15/04/2023.
+//
+
+import SwiftUI
+
+struct BottomView: View {
+    var body: some View {
+        Text(/*@START_MENU_TOKEN@*/"Hello, World!"/*@END_MENU_TOKEN@*/)
```
Comment (planecore) [inline ios/BetterRailWidget/Live Activity/Views/DynamicIsland/BottomView.swift:12]: Is this file needed?

File: ios/BetterRailWidget/Live Activity/Views/DynamicIsland/LeadingView.swift [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+struct LeadingView: View {
+  var vm: ActivityViewModel
+  
+  var headerStation: String {
+    if (vm.status == .waitForTrain || vm.status == .inExchange) {
+      return vm.train.originStationName
+    }
+    
+    return vm.train.destinationStationName
```
Comment (planecore) [inline ios/BetterRailWidget/Live Activity/Views/DynamicIsland/LeadingView.swift:15]: If my first train ends in Savidor, and my journey is to Jerusalem, Will the first text be "headed to Savidor" or "headed to Jerusalem"?

File: ios/BetterRailWidget/Live Activity/ActivityUtils.swift [hunk 4]
```diff
@@ -0,0 +1,225 @@
...
+  }
+}
+
+func getMinutesLeft(targetDate: Date) -> Int {
+  let now = Date()
+  let timeInterval = targetDate.timeIntervalSince(now)
+  let minutes = Int(round(timeInterval / 60))
+
+  if minutes <= 0 {
+    return 1
```
Comment (planecore) [inline ios/BetterRailWidget/Live Activity/ActivityUtils.swift:155]: I think if such a scenario occurs, we should show "Now".

File: app/utils/ios-helpers.tsx (hunk 1 above)
Comment (guytepper) [reply app/utils/ios-helpers.tsx:47]: > We need to modify the route the fit the original data structure, which we use in the native code

Since we override properties which type doesn't match the value we're putting inside. 
For example, `arrivalTime` is a number, but the original value that comes from the server is a string - and this is also the value which the native code expects us to pass.

I need to refactor the API response modifications in the JS side, but it's a wide project refactor so I preferred to leave it aside for later.

File: ios/BetterRailWidget/Live Activity/Views/DynamicIsland/LeadingView.swift (hunk 3 above)
Comment (guytepper) [reply ios/BetterRailWidget/Live Activity/Views/DynamicIsland/LeadingView.swift:20]: Headed to Savidor. 

I've pushed a fix so it'll display the final destination

File: ios/BetterRailWidget/Live Activity/ActivityUtils.swift (hunk 4 above)
Comment (guytepper) [reply ios/BetterRailWidget/Live Activity/ActivityUtils.swift:155]: I prefer to avoid displaying "Now", and instead to hide the counter once the status changes to `arrived` - and to keep displaying "1 min" until then. wdyt?

File: app/screens/route-details/route-details-screen.tsx [hunk 5]
```diff
@@ -141,11 +141,13 @@ export const RouteDetailsScreen = observer(function RouteDetailsScreen({ route }
             title={translate("ride.startRide")}
             loading={ride.loading}
             disabled={!isStartRideButtonEnabled}
+            onDisabledPress={() => {
+              alert(translate("ride.disabledButtonAlert"))
```
Comment (planecore) [inline app/screens/route-details/route-details-screen.tsx:145]: Use [React Native's Alert component](https://reactnative.dev/docs/alert).

File: package.json [hunk 6]
```diff
@@ -37,6 +37,7 @@
     "@react-navigation/stack": "6.3.16",
     "apisauce": "2.0.0",
     "date-fns": "^2.19.0",
+    "dayjs": "^1.11.7",
```
Comment (guytepper) [inline package.json:40]: better to use `date-fns` instead

**Review by planecore — APPROVED** [review]
